bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/obsrecord/PrimordialVisitCalculator.java

```diff
@@ -0,0 +1,253 @@
+package edu.gemini.spModel.obsrecord;
+
+import edu.gemini.pot.sp.SPObservationID;
+import edu.gemini.skycalc.Night;
+import edu.gemini.skycalc.ObservingNight;
+import edu.gemini.skycalc.TwilightBoundType;
+import edu.gemini.skycalc.TwilightBoundedNight;
+import edu.gemini.spModel.core.Site;
+import edu.gemini.spModel.dataset.DatasetLabel;
+import edu.gemini.spModel.dataset.DatasetQaState;
+import edu.gemini.spModel.event.EndDatasetEvent;
+import edu.gemini.spModel.event.ObsExecEvent;
+import edu.gemini.spModel.event.OverlapEvent;
+import edu.gemini.spModel.event.StartDatasetEvent;
+import edu.gemini.spModel.obsclass.ObsClass;
+import edu.gemini.spModel.time.ChargeClass;
+
+import java.util.List;
+import java.util.function.Function;
+import java.time.Instant;
+
+/**
+ * This is the algorithm that has been in use since the beginning of time
+ * accounting.  It is confusing, hard to follow, and suffers from several (now)
+ * known bugs:
+ *
+ * 1. It fails to charge for the portion of a dataset after a start event but
+ * before an overlap event.
+ *
+ * 2. In general, intervening dataset events that split start/end events are not
+ * charged correctly in all cases.
+ *
+ * 3. If the first event should happen to be a start dataset event then the
+ * first dataset isn't charged.
+ *
+ * Nevertheless, we cannot change time accounting for past observations so none
+ * of these bugs will be fixed in the PrimordialVisitCalculator below.  The
+ * code is essentially unchanged but instead simply lifted from `PrivateVisit`
+ * and given a calc method with fewer dependencies.
+ */
+enum PrimordialVisitCalculator {
```

Comment:
I'm ashamed, but I have to admit it was (a younger version of) me who wrote this code.  I'd hoped to delete it and replace it with a simplified version in `VisitCalculator` but I found so many bugs in it that we cannot fix without changing past results that I decided to leave it alone.  Keeping in mind that this isn't really new code, it might be best to skip it and focus on `VisitCalculator` instead.

Reply:
I'll trust you and skip it 😉